presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergSessionProperties.java

```diff
@@ -0,0 +1,401 @@
+                        "ORC: Size of buffer for streaming reads",
+                        hiveClientConfig.getOrcStreamBufferSize(),
+                        false),
+                dataSizeSessionProperty(
+                        ORC_TINY_STRIPE_THRESHOLD,
+                        "ORC: Threshold below which an ORC stripe or file will read in its entirety",
+                        hiveClientConfig.getOrcTinyStripeThreshold(),
+                        false),
+                dataSizeSessionProperty(
+                        ORC_MAX_READ_BLOCK_SIZE,
+                        "ORC: Soft max size of Presto blocks produced by ORC reader",
+                        hiveClientConfig.getOrcMaxReadBlockSize(),
+                        false),
+                booleanProperty(
+                        ORC_LAZY_READ_SMALL_RANGES,
+                        "Experimental: ORC: Read small file segments lazily",
+                        hiveClientConfig.isOrcLazyReadSmallRanges(),
+                        false),
+                booleanProperty(
+                        ORC_ZSTD_JNI_DECOMPRESSION_ENABLED,
+                        "use JNI based zstd decompression for reading ORC files",
+                        hiveClientConfig.isZstdJniDecompressionEnabled(),
+                        true),
+                dataSizeSessionProperty(
+                        ORC_STRING_STATISTICS_LIMIT,
+                        "ORC: Maximum size of string statistics; drop if exceeding",
+                        orcFileWriterConfig.getStringStatisticsLimit(),
+                        false),
+                booleanProperty(
+                        ORC_OPTIMIZED_WRITER_ENABLED,
+                        "Experimental: ORC: Enable optimized writer",
+                        hiveClientConfig.isOrcOptimizedWriterEnabled(),
+                        false),
+                booleanProperty(
+                        ORC_OPTIMIZED_WRITER_VALIDATE,
+                        "Experimental: ORC: Force all validation for files",
+                        hiveClientConfig.getOrcWriterValidationPercentage() > 0.0,
+                        false),
+                new PropertyMetadata<>(
+                        ORC_OPTIMIZED_WRITER_VALIDATE_PERCENTAGE,
+                        "Experimental: ORC: sample percentage for validation for files",
+                        DOUBLE,
+                        Double.class,
+                        hiveClientConfig.getOrcWriterValidationPercentage(),
+                        false,
+                        value -> {
+                            double doubleValue = ((Number) value).doubleValue();
+                            if (doubleValue < 0.0 || doubleValue > 100.0) {
+                                throw new PrestoException(
+                                        INVALID_SESSION_PROPERTY,
+                                        format("%s must be between 0.0 and 100.0 inclusive: %s", ORC_OPTIMIZED_WRITER_VALIDATE_PERCENTAGE, doubleValue));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format double: doubleValue (declared at line 193). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 197. Consider formatting this data with &#39;%f&#39; instead.